File: 1]. DSA/1]. Data Structures/01]. Array/JavaScript/_009)_Remove_Duplicates_from_a_sorted_array.js
function remDups(arr) {
    let n = arr.length + 1;
    let temp = [];
    for (let i = 0; i < n; i++) {
        if (arr[i] != arr[i + 1]) {
            temp.push(arr[i]);
        }
    }
    return temp;
}

let arr = [0, 20, 20, 30, 30, 30, 30];
console.log(`Before Removal Duplicates-> ${arr}`);

let Res = remDups(arr);
console.log(`After Removal Duplicates-> ${Res}`);

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/C++/_009)_Remove_Duplicates_from_a_sorted_array.cpp
#include<iostream>
using namespace std;

int remDups(int arr[], int n)
{
    int temp[n];
    temp[0] = arr[0];
    int res = 1;
    for(int i=0; i<n; i++)
    {
        if(temp[res-1] != arr[i])
        {
            temp[res] = arr[i];
            res++;
        }
    }
    for(int i=0; i<n; i++)
    {
        arr[i] = temp[i];
    }
    return res;
}

int main()
{
    int arr[] = {10, 20, 20, 30, 30, 30, 30}, n=7;
    cout<<"Before Removal Duplicates: ";
    for(int i=0; i<n; i++)
    {
        cout<<arr[i]<<"  ";
    }
    cout<<endl<<"Size = "<<n<<endl;

    int Res = remDups(arr, n);
    cout<<"After Removal Duplicates: ";
    for(int i=0; i<Res; i++)
    {
        cout<<arr[i]<<"  ";
    }
    cout<<endl<<"Size = "<<Res<<endl;
}
